test/main/modules.js

```diff
@@ -309,8 +311,14 @@ QUnit.module('QUnit.module', function () {
     QUnit.module('nested suite `this`', function (hooks) {
       this.inner = true;
 
+      hooks.before(function () {
+        this.innerBefore = 'inner before value';
+      });
+
       hooks.beforeEach(function (assert) {
         assert.strictEqual(this.outer, 2);
+        assert.strictEqual(this.outerBefore, 'outer before value');
+        assert.strictEqual(this.innerBefore, 'inner before value');
```

Comment:
should we assert all of these things in `hooks.afterEach` as well?

Reply:
Sure!